gax-java/gax-httpjson/src/main/java/com/google/api/gax/httpjson/longrunning/stub/HttpJsonOperationsStub.java

```diff
@@ -309,32 +325,113 @@ protected HttpJsonOperationsStub(
+                      .setAdditionalPaths(
+                          customOperationHttpBindings.get(LRO_GET_OPERATION)
+                              .getAdditionalBindingsList().stream()
+                              .map(HttpJsonOperationsStub::getValueBasedOnPatternCase)
+                              .toArray(String[]::new))
+                      .build())
+              .build();
+    }
+
+    if (customOperationHttpBindings.containsKey(LRO_DELETE_OPERATION)) {
+      deleteOperationMethodDescriptor =
+          deleteOperationMethodDescriptor
+              .toBuilder()
+              .setRequestFormatter(
+                  ((ProtoMessageRequestFormatter<DeleteOperationRequest>)
+                          deleteOperationMethodDescriptor.getRequestFormatter())
+                      .toBuilder()
+                      .updateRawPath(
+                          customOperationHttpBindings.get(LRO_DELETE_OPERATION).getDelete())
+                      .setAdditionalPaths(
+                          customOperationHttpBindings.get(LRO_DELETE_OPERATION)
+                              .getAdditionalBindingsList().stream()
+                              .map(HttpJsonOperationsStub::getValueBasedOnPatternCase)
+                              .toArray(String[]::new))
+                      .build())
+              .build();
     }
 
-    ApiMethodDescriptor.Builder<RequestT, ResponseT> descriptorBuilder =
-        methodDescriptor.toBuilder();
-    ProtoMessageRequestFormatter<RequestT> requestFormatter =
-        (ProtoMessageRequestFormatter<RequestT>) descriptorBuilder.getRequestFormatter();
+    if (customOperationHttpBindings.containsKey(LRO_CANCEL_OPERATION)) {
+      cancelOperationMethodDescriptor =
+          cancelOperationMethodDescriptor
+              .toBuilder()
+              .setRequestFormatter(
+                  ((ProtoMessageRequestFormatter<CancelOperationRequest>)
+                          cancelOperationMethodDescriptor.getRequestFormatter())
+                      .toBuilder()
+                      .updateRawPath(
+                          customOperationHttpBindings.get(LRO_CANCEL_OPERATION).getPost())
+                      .setAdditionalPaths(
+                          customOperationHttpBindings.get(LRO_CANCEL_OPERATION)
+                              .getAdditionalBindingsList().stream()
+                              .map(HttpJsonOperationsStub::getValueBasedOnPatternCase)
+                              .toArray(String[]::new))
+                      .build())
+              .build();
+    }
+  }
 
-    return descriptorBuilder
-        .setRequestFormatter(
-            requestFormatter.toBuilder().updateRawPath("/v1/", '/' + apiVersion + '/').build())
-        .build();
+  /* This function returns the list of method descriptors (custom or default) */
+  @InternalApi
+  public List<ApiMethodDescriptor> getAllMethodDescriptors() {
```

Comment:
We can make remove the `public` here since it's only used in tests, we should also make that clear in the comment.

Reply:
MockHttpService requires this in OperationsClientTest. I'll add `@VisibleForTesting` annotation here to mark its intended use case.